File: generate_sitemap.py
import xml.etree.ElementTree as ET
from datetime import datetime

locales = ['ar', 'en', 'fr']
base_paths = [
    '',
    '/privacy',
    '/terms',
    '/works/nagha-festival',
    '/works/maatamoulana',
    '/works/dar-mauritanie',
    '/works/cinemasa',
    '/works/pan-mauritania',
    '/works/accespan',
    '/works/cultural-map',
    '/works/bankily-ledger'
]

urlset = ET.Element('urlset', xmlns="http://www.sitemaps.org/schemas/sitemap/0.9")
today = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.000Z')

for locale in locales:
    for path in base_paths:
        url = ET.SubElement(urlset, 'url')
        loc = ET.SubElement(url, 'loc')
        loc.text = f'https://www.afrikyia.com/{locale}{path}'
        
        lastmod = ET.SubElement(url, 'lastmod')
        lastmod.text = today
        
        changefreq = ET.SubElement(url, 'changefreq')
        changefreq.text = 'monthly'
        
        priority = ET.SubElement(url, 'priority')
        if path == '':
            priority.text = '1.0'
            changefreq.text = 'yearly'
        elif path in ['/privacy', '/terms']:
            priority.text = '0.8'
        else:
            priority.text = '0.7'

# Prettify the XML
from xml.dom import minidom
xmlstr = minidom.parseString(ET.tostring(urlset)).toprettyxml(indent="  ")

with open('public/sitemap.xml', 'w') as f:
    f.write(xmlstr)

print("Sitemap generated successfully.")

File: scripts/generate-sitemap.ts
import fs from 'fs';
import path from 'path';

const achievementsPath = path.join(process.cwd(), 'src', 'data', 'achievements.json');
const achievements = JSON.parse(fs.readFileSync(achievementsPath, 'utf8'));

const baseUrl = 'https://www.afrikyia.com';
const today = new Date().toISOString();

let xml = `<?xml version="1.0" encoding="UTF-8"?>\n`;
xml += `<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">\n`;

const staticRoutes = [
    { url: '/', priority: '1.0', changefreq: 'yearly' },
    { url: '/privacy', priority: '0.8', changefreq: 'monthly' },
    { url: '/terms', priority: '0.8', changefreq: 'monthly' }
];

for (const route of staticRoutes) {
    xml += `  <url>\n`;
    xml += `    <loc>${baseUrl}${route.url === '/' ? '' : route.url}</loc>\n`;
    xml += `    <lastmod>${today}</lastmod>\n`;
    xml += `    <changefreq>${route.changefreq}</changefreq>\n`;
    xml += `    <priority>${route.priority}</priority>\n`;
    xml += `  </url>\n`;
}

for (const item of achievements) {
    xml += `  <url>\n`;
    xml += `    <loc>${baseUrl}/works/${item.id}</loc>\n`;
    xml += `    <lastmod>${today}</lastmod>\n`;
    xml += `    <changefreq>monthly</changefreq>\n`;
    xml += `    <priority>0.7</priority>\n`;
    xml += `  </url>\n`;
}

xml += `</urlset>`;

fs.writeFileSync(path.join(process.cwd(), 'public', 'sitemap.xml'), xml);
console.log('Static sitemap generated at public/sitemap.xml');
